### ISSUE TITLE
Escaped "reserved characters" - entered in text editor are replaced

### ISSUE THREAD

**Issue by clach04** [jrnl-org/jrnl#1547]
### Environment

  - `jrnl --diagnostic` output: 

        jrnl: v2.8.4
        Python: 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 22:22:05) [MSC v.1916 64 bit (AMD64)]
        OS: Windows 10

  - Install method: pipx (I was not able to install successfully any other way)

### Current Behavior

Escaped "reserved characters" are getting replaced. See steps

I spotted this when copy/pasting some source code into my journal entry.

### Expected Behavior

The NOTE section in https://github.com/jrnl-org/jrnl/blob/develop/docs/usage.md#composing-entries strongly implies using a text editor will avoid problems (but it's not 100% clear what the behavior is supposed to be.

IMHO ;-) preserving the text as entered (other than macro replacement for `yesterday:`, etc.) seems reasonable.

### Repro Steps

1. starting jrnl with no parameters to use my configured text editor and then entering:

        backslash-n \nis replaced with a newline
        backslash-n \tmight end up being a tab

2. saving results in the text as entered being removed. 

### Other Information

No runtime errors, a behavior/behaviour issue

**Comment by micahellison** [comment]
Thanks @clach04 for filing this. It's definitely a bug!

Looks like the culprit is in Entry.py _parse_text and only with `\n`. 

A resolution to this:
- [ ] change the behavior to look for newlines instead of \n
- [ ] add a test to confirm that \n isn't parsed as a newline
- [ ] add a test to confirm that splitting the title from the entry still works as expected, if there isn't one already

**Comment by pjz** [comment]
I think the issue is https://github.com/jrnl-org/jrnl/blob/develop/jrnl/Journal.py#L313 ; why is that being done?

**Comment by micahellison** [comment]
> I think the issue is https://github.com/jrnl-org/jrnl/blob/develop/jrnl/Journal.py#L313 ; why is that being done?

You're right, though I'm not sure why it's there. Still, it's not documented and it's not expected, so I think it's worth changing.
